django/db/models/fields/related_descriptors.py

```diff
@@ -87,15 +87,21 @@ class Child(Model):
     def __init__(self, field_with_rel):
         self.field = field_with_rel
 
-    @cached_property
+    @property
```

Comment:
You'll need to make sure to use `cached_property` here for performance reason and because it will create a new type on every access if `property` is used which will make `assertIs` check fails. e.g. `self.assertIs(Event.group.RelatedObjectDoesNotExist, Event.group.RelatedObjectDoesNotExist)` will fail even if pickling isn't involved.